cdap-tms/src/main/java/co/cask/cdap/messaging/store/AbstractMessageTable.java

```diff
@@ -117,13 +117,26 @@ public void delete(TopicId topicId, long startTimestamp, short startSequenceId,
     delete(startRow, Bytes.stopKeyForPrefix(stopRow));
   }
 
+  protected long decodeTx(byte[] txPtr) {
+    return Bytes.toLong(txPtr);
+  }
+
+  protected byte[] encodeTx(long txPtr) {
```

Comment:
Having utility like method in parent is not a very good idea, as it doesn't govern the behavior (e.g. when this method is used for what purpose), which the parent should do. If we need util method, It's better to have one method call `byte[] flipSign(byte[] longBytes)` instead, which we can put in `Bytes` util class. And flipping doesn't need to decode it to long.